lib/chef/resource_resolver.rb

```diff
@@ -107,7 +107,7 @@ def list
     #
     # @api private
     def provided_by?(resource_class)
-      !prioritized_handlers.include?(resource_class)
+      potential_handlers.include?(resource_class)
```

Comment:
It was wrong--prioritized_handlers is a list of classes filtered by node, while provided_by? is supposed to ignore node and just tell you whether resource_class is involved at all, on *any* node.